### DISCUSSION TITLE
label length vs kubernetes

### DISCUSSION THREAD

**Discussion by nivekuil** [containers/podman#13901] [General]
Podman labels don't have the tight constraints (63 chars) that k8s labels do. Is this intended? It's handy to rely on this to filter for pods with a certain label key through the API, which is not possible with annotations (the k8s compatible way).

**Comment by nivekuil** [comment]
I just realized that there isn't even a way to put annotations on pods in Podman. Bug? Use case is running an OCI hook, which can filter on annotation, on pod start.

**Comment by rhatdan** [comment]
Please open separate issues for each one of these.  Podman labels have no limits, but if you believe we should constrain podman play kube to only allow 63 chars and have podman generate kube truncate labels at 63 chars, with a warning, we would consider it.

**Comment by nivekuil** [comment]
On further thought podman is forced to break compatibility with something here, since docker labels are unlimited. These may map more cleanly to k8s annotations but that ship has sailed and would be confusing anyway. As such I don't really think there's a standard to follow so it seems fine that pods can't be annotated, or at least I don't have an actual problem to be solved there.

Apr 18, 2022 3:27:54 AM Daniel J Walsh ***@***.***>:

**Reply by rhatdan** [reply]
What do you think @haircommander @umohnani8 ?

**Reply by mheon** [reply]
Best solution would seem to be to limit length accepted by `podman play kube` (and warn on `podman generate kube` generating YAML with too-long labels), but do not change how `podman run` and similar handle labels.

**Reply by rhatdan** [reply]
SGTM
